File: ds/stack_test.py
from ds.stack import Stack


def test_init_stack():
    stack = Stack()
    assert stack.empty()
    assert stack.pop() is None
    assert stack.search("A") == -1


def test_push_pop_peek():
    stack = Stack()
    stack.push("A")
    assert stack.peek() == "A"
    assert stack.pop() == "A"
    assert stack.peek() is None


def test_search():
    stack = Stack()
    stack.push("A")
    stack.push("B")
    stack.push("C")
    assert stack.search("A") == 3
    assert stack.search("D") == -1

File: go/structures/stack_test.go
package structures

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestListStack(t *testing.T) {
	stack := &ListStack[int]{}
	stack.Push(1)
	assert.Equal(t, 1, stack.Size())

	got, err := stack.Pop()
	assertNilError(t, err)
	assertEqual(t, got, 1)

	stack.Push(5)
	got, err = stack.Peek()
	assertNilError(t, err)
	assertEqual(t, got, 5)

	stack.Clear()
	assertEqual(t, stack.Size(), 0)

	got, err = stack.Pop()
	assertError(t, err)
}
